aws-android-sdk-iot/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -779,20 +805,43 @@ public void onMessageArrived(String topic, byte[] data) {
 
     @Test
     public void mqttWebSocketReconnect() throws Exception {
+        websocketReconnectionTest(IAM);
+    }
 
+    @Test
+    public void testCustomAuthReconnect() throws Exception {
+        websocketReconnectionTest(CUSTOM_AUTH);
+    }
+
+    private void websocketReconnectionTest(AWSIotMqttManager.AuthenticationMode authMode) throws InterruptedException {
         final ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus> statuses = new ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus>();
         final ArrayList<String> messages = new ArrayList<String>();
 
         AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-ws-rc", Region.getRegion(Regions.US_EAST_1), endpointPrefix);
 
         mqttManager.setAutoReconnect(true);
-        // connect using WebSockets and IAM credentials
-        mqttManager.connect(credentialsProvider, new AWSIotMqttClientStatusCallback() {
-            @Override
-            public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
-                statuses.add(status);
-            }
-        });
+        if (authMode.equals(IAM)) {
+            // connect using WebSockets and IAM credentials
+            mqttManager.connect(credentialsProvider, new AWSIotMqttClientStatusCallback() {
+                @Override
+                public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
+                    statuses.add(status);
+                }
+            });
+        } else if (authMode.equals(CUSTOM_AUTH)) {
+            // connect using WebSockets and custom authentication token
+            mqttManager.connect(
+                    "token-header",
+                    "allow",
+                    "Gv85h71GT07RWo5uWQpqHlYWePN7YVY7KgiuPVEnt65+tDDWyZOlGLIDxvwC7LiBF7FbmG/coeBELKltqmrP0gjM7sIyXlrx315xG3LupoK4bmsNIzg6xxWa0n60WxhOv1p5qXDAhmYWIFuL0enMtuwpCY1W073NqZ/+iOsrJexssZiiLnvVf3z6HReEEER6TfFevrgAEUva0h9pzbUepbJ6RaOO7zZY6e/T/wSHrKl0jA4MalGReKbhdlg0HyTPfILhjorYkF7ew/8Ml+QMr/WU90YMp9Sv8SF0q7WBzbuqxpAR8vxm00rRlgrFdsIRjGFTDmrp9Fsx+xlx0aADCA==",
```

Comment:
let's not have a hard-coded token here

Reply:
Couple of thoughts on this : 
1. I don't think token signature is something that has to be obfuscated. It is similar to the AWS signature that we include in all the service calls. It is generated by signing token with the private key and is meant to be "in-transit" and purpose of signing it is to prevent any wrongdoings from a malicious third party that happens to get its hand on the request. IoT verifies the authenticity of the request by using its public key.

2. As far as token itself is concerned, there is not much one can do without the user-specific IoT endpoint. It is akin to https://github.com/aws-amplify/aws-sdk-android/blob/master/aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java#L95

Purpose of putting it like this was to emphasize the above point about its usage. I will redact these for now just to be safe and will revisit this later.

Reply:
As per our earlier discussion, I think a better way to reflect example usage would be to show them how this token is obtained with SSL encryption of token and private key. Token should be obtained exactly how a customer would obtain it and private key should be the one being encrypted.

Reply:
Token is obtained using customer's own custom authentication mechanism and I have added the process for obtaining token signature in the documentation. Sample app would be more appropriate place to show how to do that programmatically instead of the tests.

Reply:
I agree that sample app is a better place for this. Sounds good to me